src/main/java/TaskList.java

```diff
@@ -0,0 +1,290 @@
+                String detailsStr = inputStr.substring(5, inputStr.length());
+                String checkIfTodoIsEmpty = detailsStr.replaceAll("\\s", "");
+
+                if (checkIfTodoIsEmpty.length() == 0) {
+                    //That means it is empty behing todo
+                    throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+                } else {
+                    Task todo = new Todo(detailsStr);
+                    storingList.add(todo);
+                    String result = printAddingTask(todo);
+
+
+                    //store into hard disk
+                    storage.convertToHardDiskFormatAndStore(todo, "T", "");
+
+                    return result;
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+            }
+
+        } else if (inputStr.contains("event")) {
+            if (inputStr.substring(0, 5).equals("event") && inputStr.length() > 6) {
+                String[] eventAndTime = inputStr.substring(6, inputStr.length()).split(" /at ");
+
+                if (eventAndTime.length <= 1) {
+                    //not able to split string properly
+                    throw new GrapieExceptions(ErrorMsg.eventFormatError);
+                } else {
+                    Event event = new Event(eventAndTime[0], eventAndTime[1]);
+                    storingList.add(event);
+
+                    //printing
+                    String result = printAddingTask(event);
+
+                    //store into hard disk
+                    storage.convertToHardDiskFormatAndStore(event, "E", eventAndTime[1]);
+                    return result;
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+            }
+
+        } else if (inputStr.contains("deadline")) {
+            if (inputStr.substring(0, 8).equals("deadline") && inputStr.length() > 9) {
+
+                String[] eventAndTime = inputStr.substring(9, inputStr.length()).split(" /by ");
+
+                if (eventAndTime.length > 1) {
+                    Deadline deadline = new Deadline(eventAndTime[0], eventAndTime[1]);
+                    storingList.add(deadline);
+
+                    //print
+                    String result = printAddingTask(deadline);
+
+                    //store into hard disk
+                    storage.convertToHardDiskFormatAndStore(deadline, "D", eventAndTime[1]);
+                    return result;
+                } else {
+                    //"OOPS!!! Deadline in wrong format. Please use: deadline your_deadline /by YYYY-MM-DD
+                    // TTTT"
+                    throw new GrapieExceptions(ErrorMsg.deadlineFormatError);
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.emptyDescriptionError);
+            }
+        } else {
+            throw new GrapieExceptions(ErrorMsg.wakarimasenError);
+        }
+
```

Comment:
I feel this method is very lengthy and needs more abstraction.

A great option is to make this method polymorphic.
eg. addTask(Task task).
Then you should be create the task before adding it into the list, rather than handling the parsing within task list.

Reply:
In other words, TaskList shouldn't need to worry about how tasks are created.